src/components/views/login/ServerConfig.js

```diff
@@ -29,8 +29,10 @@ module.exports = React.createClass({
     propTypes: {
         onHsUrlChanged: React.PropTypes.func,
         onIsUrlChanged: React.PropTypes.func,
-        defaultHsUrl: React.PropTypes.string,
-        defaultIsUrl: React.PropTypes.string,
+        initialHsUrl: React.PropTypes.string, // whatever the current value is when we create the component
+        initialIsUrl: React.PropTypes.string, // whatever the current value is when we create the component
+        defaultHsUrl: React.PropTypes.string, // e.g. https://matrix.org
+        defaultIsUrl: React.PropTypes.string, // e.g. https://vector.im
```

Comment:
from IRL with @kegsay:
 * s/initial(.*)Url/custom$1Url/g;
 * spell out the semantics that defaultUrl takes effect if customUrl is blank in the props for ServerConfig
 * replace the "use custom server options" checkbox with two radio buttons for "use default servers ( ) v. use custom servers (*)"